pkg/clustermanager/cluster_manager.go

```diff
@@ -371,6 +374,12 @@ func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster,
 		return fmt.Errorf("error running post control plane upgrade operations: %v", err)
 	}
 
+	logger.V(3).Info("Waiting for control plane upgrade to be in progress")
+	err = c.clusterClient.WaitForControlPlaneNotReady(ctx, managementCluster, ctrlPlaneInProgressStr, newClusterSpec.Name)
```

Comment:
This would affect other providers too right? What is the reason for adding this?